tutorials/curator-llm-pii/PII-modification-Enron.ipynb

```diff
@@ -0,0 +1,1114 @@
+    "                'address': sum(1 for line in text.split('\\n') \n",
+    "                             if any(a in line.lower() for a in ['address:', 'street:', 'ave.', 'road'])),\n",
+    "                'name': sum(1 for line in text.split('\\n') \n",
+    "                          if any(n in line.lower() for n in ['name:', 'mr.', 'mrs.', 'ms.', 'dr.']))\n",
+    "            }\n",
+    "            return indicators\n",
+    "        \n",
+    "        print(\"\\nAnalyzing PII patterns...\")\n",
+    "        orig_indicators = [count_pii_indicators(text) for text in original_df['text']]\n",
+    "        async_indicators = [count_pii_indicators(text) for text in async_df['text']]\n",
+    "        sync_indicators = [count_pii_indicators(text) for text in sync_df['text']]\n",
+    "        \n",
+    "        def calc_averages(indicators):\n",
+    "            if not indicators:\n",
+    "                return {}\n",
+    "            return {k: sum(d[k] for d in indicators) / len(indicators) for k in indicators[0]}\n",
+    "        \n",
+    "        print(\"\\n=== Average PII Indicators Per Email ===\")\n",
+    "        \n",
+    "        print(\"\\nIn Original Emails:\")\n",
+    "        for k, v in calc_averages(orig_indicators).items():\n",
+    "            print(f\"  • {k.title()}: {v:.2f}\")\n",
+    "        \n",
+    "        print(\"\\nAfter Async Modification:\")\n",
+    "        for k, v in calc_averages(async_indicators).items():\n",
+    "            print(f\"  • {k.title()}: {v:.2f}\")\n",
+    "        \n",
+    "        print(\"\\nAfter Sync Modification:\")\n",
+    "        for k, v in calc_averages(sync_indicators).items():\n",
+    "            print(f\"  • {k.title()}: {v:.2f}\")\n",
+    "            \n",
+    "    except Exception as e:\n",
+    "        print(f\"An error occurred during pattern analysis: {str(e)}\")\n",
+    "        print(\"\\nDebug information:\")\n",
+    "        print(f\"Is dataset available? {'Yes' if 'dataset' in locals() else 'No'}\")\n",
+    "        print(f\"Is async_modified available? {'Yes' if 'async_modified' in locals() else 'No'}\")\n",
+    "        print(f\"Is sync_modified available? {'Yes' if 'sync_modified' in locals() else 'No'}\")\n",
+    "\n",
+    "# Run both analyses\n",
+    "print(\"Starting comprehensive PII modification analysis...\")\n",
+    "compare_pii_modifications()\n",
+    "print(\"\\nAnalyzing PII patterns across all emails...\")\n",
+    "analyze_pii_patterns()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "79a69033-df1f-41dd-8355-992113cb77d3",
+   "metadata": {},
+   "source": [
+    "## Key things to note from steps 6 and 7\n",
+    "\n",
+    "• Both modification methods successfully removed personal information\n",
+    "\n",
+    "• The async method used longer replacements (like [Employee_ID_123]) while sync used shorter ones (employee.1)\n",
```

Comment:
I don't see this anywhere in the results, although maybe I'm missing something... may be best to remove this point.